apps/examples/bundle_test/bundle_test_main.c

```diff
@@ -0,0 +1,255 @@
+void test_bundle_get_invalid(void)
+{
+	bundle *b;
+
+	TEST(NULL == bundle_get_val(NULL, "key1"));
+	TEST(EINVAL == get_errno());
+
+	b = bundle_create();
+
+	TEST(NULL == bundle_get_val(b, "key4"));
+	TEST(ENOKEY == get_errno());
+
+	TEST(NULL == bundle_get_val(NULL, "aaa"));
+	TEST(EINVAL == get_errno());
+
+	TEST(NULL == bundle_get_val(b, NULL));
+	TEST(EKEYREJECTED == get_errno());
+
+	bundle_free(b);
+	printf("test_bundle_get_invalid done\n");
+}
+
+void iter_f(const char *key, const char *val, void *data)
+{
+	char *keys[] = {"1", "2", "3"};
+	char *vals[] = {"a", "b", "c"};
+	int i;
+
+	for (i = 0; i < 3; i++) {
+		if (0 == strcmp(key, keys[i])) {
+			TEST(0 == strcmp(val, vals[i]));
+			break;
+		}
+	}
+}
+
+void test_bundle_del(void)
+{
+	bundle *b;
+
+	TEST(0 != bundle_del(NULL, "key"));
+	TEST(EINVAL == get_errno());
+
+	b = bundle_create();
+
+	TEST(0 != bundle_del(b, NULL));
+	TEST(EKEYREJECTED == get_errno());
+
+	TEST(0 != bundle_del(b, ""));
+	TEST(EKEYREJECTED == get_errno());
+
+	TEST(0 != bundle_del(b, "key"));
+	TEST(ENOKEY == get_errno());
+
+	TEST(0 == bundle_get_count(b));
+
+	bundle_add(b, "k1", "v1");
+	const char *v = bundle_get_val(b, "k1");
+	TEST(NULL != v && 0 == strcmp("v1", v));
+
+	TEST(1 == bundle_get_count(b));
+
+	TEST(0 == bundle_del(b, "k1"));
+	TEST(NULL == bundle_get_val(b, "k1"));
+	TEST(ENOKEY == get_errno());
+
+	TEST(0 == bundle_get_count(b));
+
+	bundle_free(b);
+	printf("test_bundle_del done\n");
+}
+
+void test_bundle_iterate(void)
+{
+	bundle *b;
+
+	b = bundle_create();
+
+	bundle_add(b, "1", "a");
+	bundle_add(b, "2", "b");
+	bundle_add(b, "3", "c");
+
+	bundle_iterate(b, iter_f, NULL);
+
+	bundle_free(b);
+	printf("test_bundle_iterate done\n");
+}
+
+void test_bundle_encode_decode(void)
+{
+	bundle *b1, *b2;
+	bundle_raw *r;
+	int size_r;
+
+	b1 = bundle_create();
+	bundle_add(b1, "k1", "v1");
+	bundle_add(b1, "k2", "v2");
+	bundle_encode(b1, &r, &size_r);
+
+	/* show serialized data */
+	/*
+	char *c = r;
+	for(; c < r+size_r; c++) {
+		printf("%c", *c);
+		if ('\0' == *c) printf("\n");
+	}
+	*/
+
+	b2 = bundle_decode(r, size_r);
+
+	TEST(0 == strcmp(bundle_get_val(b1, "k1"), bundle_get_val(b2, "k1")));
+	TEST(0 == strcmp(bundle_get_val(b1, "k2"), bundle_get_val(b2, "k2")));
+
+	bundle_free(b1);
+	bundle_free(b2);
+	free(r);
+	printf("test_bundle_encode_decode done\n");
+}
+
+static void _dup_check_cb(const char *k, const char *v, void *data)
+{
+	bundle *b = (bundle *)data;
+
+	TEST(0 == strcmp(bundle_get_val(b, k), v));
+}
+
+void test_bundle_dup(void)
+{
+	bundle *b1, *b2;
+	b1 = bundle_create();
+	bundle_add(b1, "k1", "v1");
+	bundle_add(b1, "k2", "v2");
+
+	b2 = bundle_dup(b1);
+
+	TEST(NULL != b2);
+	bundle_iterate(b2, _dup_check_cb, b1);
+	TEST(bundle_get_count(b1) == bundle_get_count(b2));
+
+	bundle_free(b1);
+	bundle_free(b2);
+	printf("test_bundle_dup done\n");
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int bundle_test_main(int argc, char *argv[])
```

Comment:
I think you'd better change these to testcase for supporting bundles to public API.